core/src/main/java/io/confluent/rest/auth/AuthUtil.java

```diff
@@ -0,0 +1,118 @@
+package io.confluent.rest.auth;
+
+import io.confluent.rest.RestConfig;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.eclipse.jetty.security.ConstraintMapping;
+import org.eclipse.jetty.util.security.Constraint;
+
+public final class AuthUtil {
+
+  private AuthUtil() {
+  }
+
+  public static boolean isCorsEnabled(final RestConfig restConfig) {
+    String allowedOrigins = restConfig.getString(RestConfig.ACCESS_CONTROL_ALLOW_ORIGIN_CONFIG);
+    return !allowedOrigins.trim().isEmpty();
+  }
+
+  /**
+   * Build the standard global auth constraint from standard RestConfig.
+   *
+   * <p>The valid roles is extracted from {@link RestConfig#AUTHENTICATION_ROLES_CONFIG}
+   *
+   * <p>OPTIONS requests will not require auth if
+   * {@link RestConfig#ACCESS_CONTROL_ALLOW_ORIGIN_CONFIG} is not empty.
+   *
+   * @param restConfig the rest app's config.
+   * @return the constraint mapping
+   */
+  public static ConstraintMapping createGlobalAuthConstraint(final RestConfig restConfig) {
+    final List<String> roles = restConfig.getList(RestConfig.AUTHENTICATION_ROLES_CONFIG);
+    final boolean omitOptions = isCorsEnabled(restConfig);
+    return createGlobalAuthConstraint(roles, omitOptions);
+  }
+
+  /**
+   * Build the standard global auth constraint.
+   *
+   * <p>To be authorised a user must below to at least one of the supplied {@code roles}.
+   *
+   * <p>If {@code omitOptions} is {@code true} the HTTP OPTIONS requests are excluded from
+   * auth checks, e.g. as required for CORS preflight checks.
+   *
+   * @param roles the list of valid roles
+   * @param omitOptions if {@code true}, OPTIONS requests are not subject to auth checks.
+   * @return the constraint mapping.
+   */
+  public static ConstraintMapping createGlobalAuthConstraint(
+      final List<String> roles,
+      final boolean omitOptions) {
+
+    final Constraint constraint = new Constraint();
+    constraint.setAuthenticate(true);
+    constraint.setRoles(roles.toArray(new String[0]));
+
+    final ConstraintMapping mapping = new ConstraintMapping();
+    mapping.setConstraint(constraint);
+    mapping.setMethod("*");
+    if (omitOptions) {
+      mapping.setMethodOmissions(new String[]{"OPTIONS"});
+    }
+    mapping.setPathSpec("/*");
+    return mapping;
+  }
+
+  /**
+   * Build constraints for any unsecured paths defined in standard RestConfig.
+   *
+   * @param restConfig the rest app's config.
+   * @return the list of constraint mappings.
+   */
+  public static List<ConstraintMapping> createUnsecuredConstraints(final RestConfig restConfig) {
+    List<String> unsecuredPaths = restConfig.getList(RestConfig.AUTHENTICATION_SKIP_PATHS);
+    return createUnsecuredConstraints(unsecuredPaths);
+  }
+
+  /**
+   * Build constraints for any unsecured paths.
+   *
+   * @param unsecuredPaths the list of paths that should not be secured.
+   * @return the list of constraint mappings.
+   */
+  @SuppressWarnings("WeakerAccess") // API call.
```

Comment:
what is this warning about?

Reply:
Currently on the test use this method, so you get a warning saying it can be package private.